### PR TITLE
[stable/oauth2-proxy] Possibility to add extra environment variables

### PR DESCRIPTION
**Pull request by fredsted**
#### What this PR does / why we need it:

I've used this chart successfully in a bunch of places, but at an onprem customer where all internet traffic is behind a proxy, github authorization did not work. As such, I had to add environment variables `http_proxy` and `no_proxy` in order to make github.com traffic pass through the proxy. 

As far as I can tell, there's no configuration options for oauth2-proxy that allows this except for these environment variables.

#### Checklist

- [x] [DCO](https://github.com/helm/charts/blob/master/CONTRIBUTING.md#sign-your-work) signed
- [x] Chart Version bumped
- [x] Variables are documented in the README.md
- [x] Title of the PR starts with chart name (e.g. `[stable/chart]`)

### CODE REVIEW

File: stable/oauth2-proxy/templates/deployment.yaml [hunk 1]
```diff
@@ -66,6 +66,10 @@ spec:
           - --htpasswd-file=/etc/oauth2_proxy/htpasswd/users.txt
         {{- end }}
         env:
+        {{- range $key, $value := .Values.extraEnv }}
```
Comment (davidkarlsen) [inline stable/oauth2-proxy/templates/deployment.yaml]: if this could be a toYaml block it is more versatile, e.g. can refer values in configmaps/secrets

Comment (fredsted) [reply stable/oauth2-proxy/templates/deployment.yaml]: Makes sense, thanks!

File: stable/oauth2-proxy/templates/deployment.yaml [hunk 2]
```diff
@@ -66,6 +66,9 @@ spec:
           - --htpasswd-file=/etc/oauth2_proxy/htpasswd/users.txt
         {{- end }}
         env:
+        {{- if .Values.extraEnv }}
```
Comment (Miouge1) [inline stable/oauth2-proxy/templates/deployment.yaml:69]: There is already a `extraEnv` below, what is the difference?

Comment (fredsted) [reply stable/oauth2-proxy/templates/deployment.yaml:69]: Huh. Looks like #16343 did something completely identical and was merged recently. Closing this.

**Pull request closed without merging**